### PR TITLE
[46] Added svg support with preview page

### PR DESCRIPTION
**Pull request by luiswill**

### CODE REVIEW

**Comment by luiswill** [comment]
```
gulp.task('compileSvg', function () {
	const svgSprite = require('gulp-svg-sprites');

	return gulp.src(config.global.src + config.global.resources + '/icons/*.svg')
		.pipe(svgSprite({
			baseSize: 20,
			cssFile: '../../scss/fonts/_sprite.scss',
			svg: {
				sprite: "svg/sprite.svg"
			},
			preview: false
		}))
		.pipe(gulp.dest(config.global.src + config.global.resources + '/icons/sprite'));
});
```
This gulp task is compiling all svg icons in `src/resources/icons` and building a scss file. This file will then be included in the general **styles.css** file.
Also the compiled sprite.svg file containing all icons is located in the `src/resources/icons/sprite/svg `folder

```
gulp.task('copySvgFile', function() {
	return gulp.src(config.global.src + config.global.resources + '/icons/sprite/svg/sprite.svg')
		.pipe(gulp.dest(config.global.dev + config.global.resources + '/svg'));
});
```

This task is copying the compiled svg file to the `.tmp` folder so that we can reference it on the live website.


Do not hesitate if there are any questions. I also built a preview page for displaying the icons.

**Tested on IE10&11, Chrome and Mozilla.**

File: tasks/sprite.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+				sprite: "svg/sprite.svg"
+			},
+			preview: false
+		}))
+		.pipe(gulp.dest(config.global.src + config.global.resources + '/icons/sprite'));
+});
+
+gulp.task('copySvgFile', function() {
+	return gulp.src(config.global.src + config.global.resources + '/icons/sprite/svg/sprite.svg')
+		.pipe(gulp.dest(config.global.dev + config.global.resources + '/svg'));
```
Comment (janrembold) [inline tasks/sprite.js]: Why do wen need another copy task? This should be done all at once inside the compileSvg task

File: tasks/sprite.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const gulp = require('gulp');
+const config = require('./../config');
+
+gulp.task('convertIconsToSprites', function() {
```
Comment (janrembold) [inline tasks/sprite.js]: This extra task is not necessary if everything is handled inside a single task - see comment below in the copy task

File: tasks/sprite.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	const runSequence = require('run-sequence');
+
+	runSequence('compileSvg', 'copySvgFile');
+})
+
+gulp.task('compileSvg', function () {
+	const svgSprite = require('gulp-svg-sprites');
+
+	return gulp.src(config.global.src + config.global.resources + '/icons/*.svg')
```
Comment (janrembold) [inline tasks/sprite.js]: please use pump syntax like here: https://github.com/biotope/biotope-build/blob/master/pumps/uglify.js

File: tasks/sprite.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+		.pipe(svgSprite({
```
Comment (janrembold) [inline tasks/sprite.js]: the config should be in config file in config.svgSprite.options = {...}

File: tasks/sprite.js (hunk 3 above)
Comment (janrembold) [inline tasks/sprite.js]: please use path module for all paths like here: https://github.com/biotope/biotope-build/blob/master/tasks/hb2.js#L14

**Pull request closed without merging**